### DISCUSSION TITLE
Support environments() on scheduled task definition

### DISCUSSION THREAD

**Discussion by joekeilty-oub** [spatie/laravel-schedule-monitor#49] [Ideas]
Hey there, first of all thanks for your package it is very helpful.

I have a feature request to support the [environments](https://laravel.com/docs/8.x/scheduling#environment-constraints) option when defining a scheduled task.

With the setup:

.env
```
APP_ENV=staging
```

Kernel.php
```
protected function schedule(Schedule $schedule): void
{
    $schedule->command('normalcommand')->daily();
    $schedule->command('important-prod-only-command')->environments(['production'])->hourly();
}
```

When running `schedule-monitor:sync` and `schedule-monitor:list` on the staging environment, I'd expect only to monitor one command (`normalcommand`) especially if using OhDear integration, but it actually adds both of the commands even though we've constrained the second one to the production environment. This can lead to developer confusion, unexpected behaviour and potentially false positives on OhDear alerts

A workaround for this is to not use [environments](https://laravel.com/docs/8.x/scheduling#environment-constraints) option and wrap the scheduled task definitions in `if (App::environment() === 'production')` etc. checks, but I find using `->environments()` is a lot cleaner and saves duplication of code.

The `Event` class has a `runsInEnvironment()` method which could support this behaviour

**Comment by mennorenkens** [comment]
This would be nice :), using the same workaround now.

**Reply by mennorenkens** [reply]
Made a pull request @joekeilty-oub: https://github.com/spatie/laravel-schedule-monitor/pull/64

**Reply by joekeilty-oub** [reply]
Nice one! Hope it makes it's way in :)

**Reply by mennorenkens** [reply]
It's merged and a new release has already been created! 🙏
